test/conformance/volumes_test.go

```diff
@@ -0,0 +1,178 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestConfigMapVolume tests that we echo back the appropriate text from the ConfigMap volume.
+func TestConfigMapVolume(t *testing.T) {
+	clients := setup(t)
+
+	// Add test case specific name to its own logger.
+	logger := logging.GetContextLogger(t.Name())
+
+	names := test.ResourceNames{
+		Service: test.AppendRandomString("cm-volume-", logger),
+		Image:   "hellovolume",
+	}
+
+	text := test.AppendRandomString("hello-volumes-", logger)
+
+	// Create the ConfigMap with random text.
+	configMap, err := clients.KubeClient.Kube.CoreV1().ConfigMaps(test.ServingNamespace).Create(&corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: text,
+		},
+		Data: map[string]string{
+			filepath.Base(test.HelloVolumePath): text,
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	logger.Info("Successfully created configMap: %v", configMap)
+
+	cleanup := func() {
+		tearDown(clients, names)
```

Comment:
`tearDown` uses the clients to remove the resources named by `names`, the clients themselves don't need any tear down.  This order is probably better because we aren't deleting a configmap or secret out from under our resource.